### PR TITLE
fix for cases when template is not detected

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cid/helpers/quicksight/dashboard.py [hunk 1]
```diff
@@ -40,6 +40,18 @@ def deployedTemplate(self) -> CidQsTemplate:
     def deployedTemplate(self, template: CidQsTemplate) -> None:
         self._deployedTemplate = template
 
+    @property
+    def template_id(self) -> str:
+        if isinstance(self.deployedTemplate, CidQsTemplate):
+            return self.deployedTemplate.arn
```
Comment (darken99) [inline cid/helpers/quicksight/dashboard.py:46]: Shoudn't we use `self.deployedTemplate.id` here?

File: cid/helpers/quicksight/dashboard.py [hunk 2]
```diff
@@ -53,7 +65,8 @@ def latest(self) -> bool:
 
     @property
     def latest_version(self) -> int:
-        return self.sourceTemplate.version
+        if isinstance(self.sourceTemplate, CidQsTemplate):
+            return self.sourceTemplate.version
```
Comment (darken99) [inline cid/helpers/quicksight/dashboard.py:69]: What if not an instance of `CidQsTemplate`? Should we return `None` or `-1`?

**Review by darken99 — CHANGES_REQUESTED** [review]

File: cid/helpers/quicksight/dashboard.py (hunk 1 above)
Comment (iakov-aws) [reply cid/helpers/quicksight/dashboard.py]: indeed

File: cid/helpers/quicksight/dashboard.py (hunk 2 above)
Comment (iakov-aws) [reply cid/helpers/quicksight/dashboard.py:69]: No sure what is the purpose of -1 but we we can do that for consistency

**Review by darken99 — APPROVED** [review]

**Pull request merged**